c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -161,10 +162,62 @@ impl<'tcx> TypeChecker<'tcx, '_> {
         lty
     }
 
-    pub fn visit_operand(&self, op: &Operand<'tcx>) -> LTy<'tcx> {
+    pub fn visit_operand(&mut self, op: &Operand<'tcx>) -> LTy<'tcx> {
         match *op {
             Operand::Copy(pl) | Operand::Move(pl) => self.visit_place(pl),
             Operand::Constant(ref c) => {
+                if c.ty().is_any_ptr() {
+                    if let Some(alloc_id) = const_alloc_id(c) {
+                        if let Some(did) = find_static_for_alloc(&self.acx.tcx(), alloc_id) {
```

Comment:
I think we should `error!`/`panic!` if `find_static_for_alloc` fails.  If `c` has pointer/reference type, then either it should be null/dangling, in which case `const_alloc_id` will return `None`, or else it should be a pointer to a valid static, in which case `find_static_for_alloc` should return `Some`.

Reply:
this will fail in cases of string literals I believe, so I'm not sure if this makes sense to do, unless you think it needs to be covered? touched upon this a little in [this comment](https://github.com/immunant/c2rust/pull/975#discussion_r1254559773)

Reply:
Ah, good point.  Never mind, then.